source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -65,18 +65,26 @@ Filter::StreamOpenState Filter::openStream() {
   return StreamOpenState::Ok;
 }
 
-void Filter::onDestroy() {
-  ENVOY_LOG(trace, "onDestroy");
-  // Make doubly-sure we no longer use the stream, as
-  // per the filter contract.
-  processing_complete_ = true;
+void Filter::closeStream() {
   if (stream_) {
+    ENVOY_LOG(debug, "Calling close on stream");
     if (stream_->close()) {
       stats_.streams_closed_.inc();
     }
+    stream_.reset();
```

Comment:
We are actually on the call stack of RawAsyncStream here, the reason why we can free the AsyncStream stream_ is that AsyncStream doesn't own the RawAsyncStream, and the RawAsyncStream will call cleanup() later to schedule a defer delete of itself. maybe document this hack.